### PR TITLE
Upgrade marshmallow

### PR DESCRIPTION
**Pull request by Glignos**

### CODE REVIEW

File: invenio_records_rest/schemas/fields/generated.py [hunk 1]
```diff
@@ -42,6 +45,28 @@ def __init__(self, *args, **kwargs):
         # Setting "missing" to some value forces the call to ``.deserialize``
         kwargs['missing'] = GeneratedValue
         super(ForcedFieldDeserializeMixin, self).__init__(*args, **kwargs)
+    if marshmallow_version[0] >= 3:
```
Comment (slint) [inline invenio_records_rest/schemas/fields/generated.py]: Add a comment here as well explaining why we need this kind of behaviour of `deserialize` and where does it come from

File: invenio_records_rest/schemas/json.py [hunk 2]
```diff
@@ -74,21 +76,50 @@ def load_unknown_fields(self, data, original_data):
...
+                data.setdefault(pid_field, pid_value)
+            return data
+
+else:
+    class RecordMetadataSchemaJSONV1(Schema):
+        """Schema for records metadata v1 in JSON with injected PID value."""
+
+        if marshmallow_version[0] >= 3:
+            class Meta:
+                """."""
```
Comment (slint) [inline invenio_records_rest/schemas/json.py]: Add some docstring here like `Meta attributes for the schema`

File: tests/test_custom_fields.py [hunk 3]
```diff
@@ -11,17 +11,33 @@
...
+
+class CustomFieldSchema(schema_to_use):
     """Test schema."""
 
+    if marshmallow_version[0] >= 3:
+
+        class Meta:
+            """."""
+
+            from marshmallow import EXCLUDE
```
Comment (slint) [inline tests/test_custom_fields.py]: Ditto

File: tests/test_marshmallow_loader.py [hunk 4]
```diff
@@ -128,14 +131,25 @@ def test_marshmallow_load_nested_errors(app, db, es, test_data, search_url,
             headers=HEADERS)
         assert res.status_code == 400
         response_json = json.loads(res.data.decode('utf-8'))
-        assert response_json['errors'][0]['field'] == 'nested_field'
+        assert \
+            response_json['errors'][0]['field'] == 'nested_field' \
```
Comment (slint) [inline tests/test_marshmallow_loader.py]:
```suggestion
            response_json['errors'][0]['field'] in ('nested_field', 'nested')
```

File: tests/test_views_search.py [hunk 5]
```diff
@@ -314,8 +314,12 @@ def test_from_parameter_invalid_pagination(
         data = get_json(res)
         assert res.status_code == 400
         assert data['message'] == 'Invalid pagination parameters.'
-        assert {(e['field'], e['message']) for e in data['errors']} == \
-            {('from', 'Must be at least 1.'), }
+        assert \
+            {(e['field'], e['message']) for e in data['errors']} == \
```
Comment (slint) [inline tests/test_views_search.py]: To avoid duplication, store the value somewhere or use a `... in (<list-of-options>)`

**Comment by kprzerwa** [comment]
works with invenio-circulation tests 👍

**Pull request merged**